draft-ietf-quic-recovery.md

```diff
@@ -690,10 +695,23 @@ Pseudocode for OnAckReceived and UpdateRtt follow:
     if (sent_packets[ack.largest_acked]):
       latest_rtt = now - sent_packets[ack.largest_acked].time
       UpdateRtt(latest_rtt, ack.ack_delay)
-    // Find all newly acked packets.
+    // Find all newly acked packets and track the smallest
+    smallest_newly_acked = kInvalidPacketNumber
     for acked_packet in DetermineNewlyAckedPackets():
+      smallest_newly_acked =
+        min(smallest_newly_acked, acked_packet.packet_number)
```

Comment:
This assumes an ordering in the processing of the acked packets data structure. I would make this simpler:

// Find all newly acked packets in this ACK frame
newly_acked = DetermineNewlyAckedPackets(ack)

// Find the smallest packet that is newly acked in this ACK frame.
smallest_newly_acked = FindSmallestNewlyAcked(newly_acked)

You don't have to define either function, but add a comment that describes what they do.

Reply:
I don't believe it does assume ordering, that's why I'm using a min, but your suggestion works as well.